crates/base/src/vector/bvecf32.rs

```diff
@@ -0,0 +1,153 @@
+use super::{VectorBorrowed, VectorOwned};
+use crate::scalar::F32;
+use serde::{Deserialize, Serialize};
+
+pub const BVEC_WIDTH: usize = std::mem::size_of::<usize>() * 8;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct BVecf32Owned {
```

Comment:
We can add a new scalar type `bit` (boolean).

Reply:
I don't understand. Where can we add a new scalar type `bit`?

Reply:
But `+` and `-` have different meaning for `bit` and `float`. `bit` seems not unsuitable for `scalar` abstraction.